src/Datadog.Trace.ClrProfiler.Managed/Integrations/GraphQLIntegration.cs

```diff
@@ -0,0 +1,381 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+using Datadog.Trace.ClrProfiler.Emit;
+using Datadog.Trace.ClrProfiler.Helpers;
+using Datadog.Trace.Logging;
+
+namespace Datadog.Trace.ClrProfiler.Integrations
+{
+    /// <summary>
+    /// Tracing integration for GraphQL.Server.Transports.AspNetCore
+    /// </summary>
+    public static class GraphQLIntegration
+    {
+        private const string IntegrationName = "GraphQL";
+        private const string ServiceName = "graphql";
+
+        private const string ParseOperationName = "graphql.parse"; // Instrumentation not yet implemented
+        private const string ValidateOperationName = "graphql.validate";
+        private const string ExecuteOperationName = "graphql.execute";
+        private const string ResolveOperationName = "graphql.resolve"; // Instrumentation not yet implemented
+
+        private const string GraphQLAssemblyName = "GraphQL";
+        private const string GraphQLDocumentValidatorInterfaceName = "GraphQL.Validation.IDocumentValidator";
+        private const string GraphQLExecutionResultName = "GraphQL.ExecutionResult";
+        private const string GraphQLExecutionStrategyInterfaceName = "GraphQL.Execution.IExecutionStrategy";
+        private const string GraphQLValidationResultInterfaceName = "GraphQL.Validation.IValidationResult";
+
+        private const string TaskOfGraphQLExecutionResult = "System.Threading.Tasks.Task`1<" + GraphQLExecutionResultName + ">";
+
+        private static readonly ILog Log = LogProvider.GetLogger(typeof(GraphQLIntegration));
+
+        /// <summary>
+        /// Wrap the original method by adding instrumentation code around it.
+        /// </summary>
+        /// <param name="documentValidator">The instance of GraphQL.Validation.IDocumentValidator.</param>
+        /// <param name="originalQuery">The source of the original GraphQL query.</param>
+        /// <param name="schema">The GraphQL schema.</param>
+        /// <param name="document">The GraphQL document.</param>
+        /// <param name="rules">The list of validation rules.</param>
+        /// <param name="userContext">The user context.</param>
+        /// <param name="inputs">The input variables.</param>
+        /// <param name="opCode">The OpCode used in the original method call.</param>
+        /// <param name="mdToken">The mdToken of the original method call.</param>
+        /// <returns>The original method's return value.</returns>
+        [InterceptMethod(
```

Comment:
Should we add the supported assembly versions?

Reply:
Yes, I'm limiting it to GraphQL 2.3.0+. Fixed in the latest commit.